benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -544,19 +535,37 @@ void GraphicsBenchmarkApp::SetupSkyBoxMeshes()
 
 void GraphicsBenchmarkApp::SetupSphereMeshes()
 {
-    OrderedGrid grid(kMaxSphereInstanceCount, kSeed);
+    const uint32_t requiredSphereCount = std::max<uint32_t>(pSphereInstanceCount->GetValue(), kDefaultSphereInstanceCount);
+    const uint32_t initSphereCount     = std::min<uint32_t>(kMaxSphereInstanceCount, 2 * std::max(requiredSphereCount, mInitializedSpheres));
+
+    OrderedGrid grid(initSphereCount, kSeed);
+    mInitializedSpheres = initSphereCount;
 
     // Create the meshes
     uint32_t meshIndex = 0;
     for (const auto& lod : kAvailableLODs) {
         PPX_LOG_INFO("LOD: " << lod.name);
         SphereMesh sphereMesh(/* radius = */ 1, lod.value.longitudeSegments, lod.value.latitudeSegments);
         sphereMesh.ApplyGrid(grid);
-
         // Create a giant vertex buffer for each vb type to accommodate all copies of the sphere mesh
+        if (mSphereMeshes[meshIndex] != nullptr) {
```

Comment:
Is there a case where not all meshes are initialized?
I think it would be similar to just add this before the loop:

```
for (auto& mesh : mSphereMeshes)
    GetDevice()->DestroyMesh(mesh);
```

Reply:
I think at the beginning all meshes are nullptr,. Updated.